projects/fal/src/fal/app.py

```diff
@@ -300,7 +371,44 @@ async def set_global_object_preference(request, call_next):
                     "Failed set a global lifecycle preference %s",
                     self.__class__.__name__,
                 )
-            return await call_next(request)
+
+            request_id = request.headers.get(REQUEST_ID_KEY)
+            if request_id is not None:
+                await _set_logger_labels({"fal_request_id": request_id})
+
+            response = await call_next(request)
+
+            await _set_logger_labels({})
+
+            return response
+
+        @app.middleware("http")
+        async def set_request_id(request, call_next):
+            if self.isolate_channel is None:
+                async with assure_isolate_channel() as channel:
+                    self.isolate_channel = channel
```

Comment:
I think this finishes the with block and closes the channel.

```suggestion
            if self.isolate_channel is None:
                self.isolate_channel = await IsolateChannel(f"localhost:{GRPC_PORT}")
```

This seems more straight forward to me

Reply:
and we can get rid of the `assure_isolate_channel` entirely